docs/core-concepts/libraries/libraries-with-cli.md

```diff
@@ -331,186 +329,261 @@ using System.Threading.Tasks;
+You can use these entries in the `project.json` to control the name and version of the project.  If you don't specify these, the default configuration is to use the name of the containing folder as the name and 1.0.0 as the version number.
 
-```javascript
+The `project.json` files for both **AwesomeLibrary.CSharp** and **AwesomeLibrary.FSharp** now need to reference **AwesomeLibrary.Core** as a `project` target.  If you aren't multitargeting, you can use the global `dependencies` entry:
+
+```json
 {
     "dependencies":{
-        "MyLibrary.Core":""
-        ...
+        "AwesomeLibrary.Core":{
+            "version":"1.0.0",
+            "target":"project"
+        }
     }
-    ...
 }
 ```
 
-To build it all, you would open a command line at the root of the project (`/MyLibrary`).
+> **Note:** Failure to list the reference as a `project` target may result in NuGet resolving the dependency with an existing NuGet package which happens to have the same name.  Always specify `"target":"project"` when referencing a project in the same solution.
 
-Using DNX:
+If you are multitargeting, you may not be able to use a global `dependencies` entry and may have to reference **AwesomeLibrary.Core** in a target-level `dependencies` entry.  For example, if you were targeting `netstandard1.5`, you could do so like this:
 
-```
-$ dnu restore
-$ cd ../MyLibrary.CSharp
-$ dnu build
-$ cd ../MyLibrary.FSharp
-$ dnu build
-```
-    
-Using .NET CLI:
-   
-```
-$ dotnet restore
-$ cd ../MyLibrary.CSharp
-$ dotnet build
-$ cd ../MyLibrary.FSharp
-$ dotnet build
+```json
+{
+    "frameworks":{
+        "netstandard1.5":{
+            "dependencies":{
+                "AwesomeLibrary.Core":{
+                    "version":"1.0.0",
+                    "target":"project"
+                }
+            }
+        }
+    }
+}
 ```
 
-Note that `MyLibrary.Core` is automatically built because it was listed as a dependency.
+And that's it!  To learn more about `project.json` and all of the configuration you can do, check out the [.NET Core project system](../../project-model/overview.md).
 
-Any source files which use functionality in `MyLibrary.Core` can now just use it:
+## How to use native dependencies
 
-C#:
+You may wish to write a library which depends on a native `.dll`.  You have two options.  The first is to reference the native `.dll` directly in your `project.json`.  The second option is to package that `.dll` into its own NuGet package and depend on that package.
 
-```csharp
-using MyLibrary.Core;
-```
-F#:
+For the first option, you'll need to include two things:
+
+1. Setting `allowUnsafe` to `true` in a `buildOptions` section of your `project.json`.
+2. Specifying the path to the native `.dll`(s) with a [Runtime Identifier (RID)](../rid-catalog.md) under `files` in the `packOptions` section of your `project.json`.
+
+- Link to NuGet docs showing what the package looks like (https://docs.nuget.org/create/uwp-create)
```

Comment:
This links seems lost in the middle of the section. Add more context to it like: For more information about the package structure, see ...